### PR TITLE
feat: Add LLM integration usage to organization reports

### PR DESCRIPTION
**Pull request by Lior539**
## Problem

We want to know which companies are using our [LLM integrations](https://posthog.com/docs/ai-engineering/llm-insights) in PostHog. Currently, there's no way to do this

## Changes

Add the following fields to the organization_usage_report 
- event_count_from_helicone_in_period
- event_count_from_langfuse_in_period
- event_count_from_keywords_ai_in_period
- event_count_from_traceloop_in_period

## Does this work well for both Cloud and self-hosted?

No impact

## How did you test this code?

Added tests + ran `pytest posthog/tasks/test/test_usage_report.py`

### CODE REVIEW

**Comment by raquelmsmith** [comment]
I'm really sorry but I'm nervous about this being 4 extra big queries across the events table. I will look tomorrow to see if I can do a rollup. If not I will merge this and hope it's okay.

**Comment by Lior539** [comment]
@raquelmsmith with the help of ChatGPT I managed to combine it into a single query. Let me know if this looks better!

File: posthog/tasks/usage_report.py [hunk 1]
```diff
@@ -662,6 +702,20 @@ def _get_all_usage_data(period_start: datetime, period_end: datetime) -> dict[st
...
+    # Process the integration event counts
+    teams_with_event_count = {
+        "helicone": defaultdict(int),
+        "langfuse": defaultdict(int),
+        "keywords_ai": defaultdict(int),
+        "traceloop": defaultdict(int),
+    }
+
+    for team_id, integration, count in integration_event_counts:
+        teams_with_event_count[integration][team_id] = count
```
Comment (raquelmsmith) [inline posthog/tasks/usage_report.py:717]: I'd put this processing inside `get_teams_with_llm_integration_event_counts_in_period` so it's just called from the return statement like all the others.

File: posthog/tasks/usage_report.py [hunk 2]
```diff
@@ -650,10 +688,12 @@ def convert_team_usage_rows_to_dict(rows: list[Union[dict, tuple[int, int]]]) ->
...
             # Some queries return a dict with team_id and total
             team_id_map[row["team_id"]] = row["total"]
-        else:
+        elif isinstance(row, list | tuple) and len(row) == 2:
             # Others are just a tuple with team_id and total
             team_id_map[int(row[0])] = row[1]
-
+        else:
+            # Handle the case where row is an integer
+            team_id_map[row] = 1  # or some other default value
```
Comment (raquelmsmith) [inline posthog/tasks/usage_report.py:696]:
```suggestion
            team_id_map[row] = 1
```

Is this the default we want? When would we end up with this default?

File: posthog/tasks/usage_report.py (hunk 1 above)
Comment (raquelmsmith) [reply posthog/tasks/usage_report.py:717]: Oh, I see why you did this, so you aren't calling the function multiple times. I think `get_teams_with_llm_integration_event_counts_in_period` can return `teams_with_event_count["helicone"]` etc so you just call it once like you do up here, and then do `integration_event_counts["helicone"] in the return statement.

**Review by raquelmsmith — COMMENTED** [review]
Nice work with this! Thank you for taking it on 🙏 One suggestion for clarity, one question, and the type annotation should be fixed. but otherwise looks great 🙌

File: posthog/tasks/usage_report.py (hunk 1 above)
Comment (Lior539) [reply posthog/tasks/usage_report.py:717]: Updated!

**Comment by raquelmsmith** [comment]
@Lior539 can you fix the code quality check failures?

**Comment by Lior539** [comment]
@raquelmsmith fixed the code  quality checks :)

File: posthog/tasks/usage_report.py (hunk 2 above)
Comment (Lior539) [reply posthog/tasks/usage_report.py]: managed to fix it so this is no longer needed!

**Review by raquelmsmith — APPROVED** [review]

**Pull request merged**